src/main/java/org/alfresco/heartbeat/HBDataCollectorServiceImpl.java

```diff
@@ -225,24 +226,31 @@ private synchronized void scheduleCollector(HBBaseDataCollector collector) throw
         }
         else
         {
-            if (logger.isDebugEnabled())
-            {
-                logger.debug("heartbeat job unscheduled");
-            }
-            scheduler.unscheduleJob(triggerName, Scheduler.DEFAULT_GROUP);
+            unscheduleJob(triggerName, collector);
         }
     }
 
+    private void unscheduleJob(String triggerName, HBBaseDataCollector collector) throws SchedulerException
+    {
+        if (logger.isDebugEnabled())
+        {
+            logger.debug("heartbeat job unscheduled with collector: " + collector.getCollectorId() + " " + collector.getCollectorVersion());
+        }
+        scheduler.unscheduleJob(triggerName, Scheduler.DEFAULT_GROUP);
+    }
```

Comment:
Do the logging after calling scheduler.unscheduleJob(triggerName, Scheduler.DEFAULT_GROUP); or change the message. As it is now it will log that heart beat job was unscheduled before actually unscheduling so if this throws an exception the log will say that the job is unscheduled when in fact it isn't